File: src/services/instructions/loadstore/ST_W_MXX_REG.java

 import Instruction from '../Instruction'
import { REGISTER_VALUE_NAME_MAPPER } from '@/constants/registers'

const ASSEMBLER_INSTRUCTION_EXPRESSION = (sdestination, ssource) => `ld.s ${sdestination}, [${ssource}]`

export default class ST_W_MXX_REG extends Instruction {
	public ST_W_MXX_REG(memory, address, source, 
 destination) {
		super(memory, address, source, destination)
		super.setArgument32()
		super.setAssembler("st.w [0x%08x]" + ", " + this.sdest)
	}

	
	exec ({ context, memory }) {
		setMemContent(context, Instruction.fix(this.argument) / 2, context[REGISTER_VALUE_NAME_MAPPER[this.source]] , Instruction.fix(this.argument))
		
		context.pc  += 6
		updateViewer32(context, Instruction.fix(this.argument), context[REGISTER_VALUE_NAME_MAPPER[this.destination]] )
	}
}

File: src/services/instructions/loadstore/StWMXXReg.js
import Instruction from '../Instruction'
import { REGISTER_VALUE_NAME_MAPPER } from '@/constants/registers'

const ASSEMBLER_INSTRUCTION_EXPRESSION = (sdestination) => `st.w %s, ${sdestination}`

export default class StWMXXReg extends Instruction {
	constructor (memory, address, source, destination, symbolTable) {
		super(memory, address, source, destination, symbolTable)
		super.setArgument32(memory)
		super.setAssembler(ASSEMBLER_INSTRUCTION_EXPRESSION(this.sdestination), symbolTable)
	}

	exec ({ context }) {
		Instruction.setMemContent(
			context,
			Instruction.fix(this.argument) / 2,
			context[REGISTER_VALUE_NAME_MAPPER[this.source]],
			Instruction.fix(this.argument)
		)

		context.pc  += 6

		return {
			address: Instruction.fix(this.argument),
			content: context[REGISTER_VALUE_NAME_MAPPER[this.destination]]
		}
	}
}
